osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -71,6 +73,25 @@ protected override void LoadComplete()
             void next() => Next();
         }
 
+        public override void OnSuspending(ScreenTransitionEvent e)
+        {
+            var backgroundScreenStack = Parent as BackgroundScreenStack;
+            Debug.Assert(backgroundScreenStack != null);
+
+            if (background is BeatmapBackgroundWithStoryboard storyboardBackground)
+                storyboardBackground.UnloadStoryboard(backgroundScreenStack.ScheduleToTransitionEnd);
```

Comment:
This feels backwards. Why is the background receiving a delegate from the stack rather than the other way around? i.e. `backgroundScreenStack.ScheduleToTransitionEnd(storyboardBackground.UnloadStoryboard)`.

Review comment is applicable only when it is explained what the `ScheduleToTransitionEnd()` contraption is and why it is necessary as per the other review comment, because I'd rather it did not exist.

Reply:
There [used to be](https://github.com/ppy/osu/pull/25400/commits/e9471589697083d8ea2214b299aca226085e7de9) a fade out transition that was added earlier in this PR inside `UnloadStoryboard`, but now that it's removed, I can do it the other way around.

Reply:
Applied in https://github.com/ppy/osu/pull/25400/commits/080f13e34d0f5f60381d813edfd590aa2b466194